src/MediatR/IStreamRequestHandler.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+namespace MediatR
+{
+    /// <summary>
+    /// Defines a handler for a stream request using IAsyncEnumerable as return type.
+    /// </summary>
+    /// <typeparam name="TRequest">The type of request being handled</typeparam>
+    /// <typeparam name="TResponse">The type of response from the handler</typeparam>
+    public interface IStreamRequestHandler<in TRequest, out TResponse>
+        where TRequest : IRequest<TResponse>
+    {
+        /// <summary>
+        /// Handles a stream request with IAsyncEnumerable as return type.
+        /// </summary>
+        /// <param name="request">The request</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>Response from the request</returns>
+        IAsyncEnumerable<TResponse> HandleAsync(TRequest request, CancellationToken cancellationToken);
```

Comment:
For consistency with the other handler interfaces consider using just Handle rather than HandleAsync

Reply:
Renamed all HandleAsync to Handle.
Thanks for feedback!